tests/tasks/test_control_flow.py

```diff
@@ -164,6 +164,107 @@ def test_merge_imperative_flow():
     assert state.result[c].result == 2
 
 
+def test_mapped_switch_and_merge_functional():
+    with Flow("iterated map") as flow:
+        mapped_result = identity.copy().map(["a", "b", "c"])
+
+        a = identity.copy().bind("a")
+        b = identity.copy().bind("b")
+        c = identity.copy().bind("c")
+
+        switch(condition=mapped_result, cases=dict(a=a, b=b, c=c), mapped=True)
+
+        merge_result = merge(a, b, c, flow=flow, mapped=True)
```

Comment:
No need to pass in the flow, this is inferred from the context. The `flow` kwarg is there to support using `merge` when not under a `Flow` context manager.